### ISSUE TITLE
Construct Basic Sitemap

### ISSUE THREAD

**Issue by daviddarnes** [daviddarnes/bread.li#43]
Bread.li is planned to be a directory site of CMSs from all over the web, in all sizes and platforms. The plan for this issue is to flesh out the page structure and establish all the key ones needed to get this site going. I've listed my first iteration below:

- Front page
- Languages
- Frameworks
- Type
- Blog

**Comment by daviddarnes** [comment]
Need to establish pages and what they contain. Discussion is welcome!

**Comment by daviddarnes** [comment]
Thinking about this more there isn't a reason to have these main nav items until the categories and tags are sorted.

- **Blog** will be managed with the standard blogging tools within Jekyll, so this works just in the same way blogs normally work.

- **CMSs** will be a custom [collection](http://jekyllrb.com/docs/collections/) as it is known in Jekyll language. I think collections is being expanded on in future versions of Jekyll, so this should get more powerful over time. However this isn't a major concern, I just want to be able to list them and be searchable.

- **Submit** is something I've just thought of. This will probably be quite important and should link to the GitHub repo, it's the best way to submit as people can literally write the post and make a Pull Request to the repo. This technique is currently being used perfectly on [Beautiful Open](https://github.com/trek/beautiful-open/blob/gh-pages/CONTRIBUTING.md#new-pull-requests).

- **About**, may not be looked at much but is certainly needed to explain the site in a bit more detail. Maybe explain about how it was built and the videos behind it.

- **The front page** I think will be the place where all CMSs that are added will be shown, cut right to the chase. There maybe a later date where we want to show the latest popular ones but that can't really be considered until people are actually taking an interest in the website.

Any thoughts on this? Any you'd add or remove, maybe what you would like to see on these pages?

**Comment by nouveller** [comment]
Yo.

I like the submission process, nice idea. As long as it's well documented in the `README` it should work well.

I'm not sure how jobs come into it though? If Bread.li is designed to be a directory of CMS's and if it's doing that well, I don't see the need for a job board. There are plenty of sights tailored for this already, this could muddy the water.

As an aside, the most useful thing will be to find CMS's by language, features, ease of use/installation.

**Comment by daviddarnes** [comment]
@nouveller sights? heh you make a good point, I just retweet so many that I thought I could just chuck embedded tweets on there and leave it at that. But yes, there are a lot of jobs boards out there at the moment.

Filtering options would you like to see then?
- Languages
- Hosting Type
- Content Type

**Comment by nouveller** [comment]
Unless they are CMS related, that may be some added value? e.g. "WordPress are hiring a front-end dev".

I was just thinking about what kind of things I'd think about to make a decision about a CMS. There are a couple of viewpoints too, from a front-end, back-end and client point of view.

For clients, I'd want something to indicate how easy to use the admin side of things is. For front-enders, how easy it is to integrate templates and for the back-enders, how easy is it to setup etc...

But they are probably quite low level requirements, so maybe not suitable for filters.

Certainly:

- Language (PHP, Ruby etc..)
- Cost (Free or paid for)
- Hosted (self or on a platform)
- Content type (shop, blog, etc...)
- Rating (to determine most popular)

**Comment by daviddarnes** [comment]
@nouveller Ah yes, keep it closely related to the content. Front-end devs, web designers, *name of CMS* developer and alike.

Those are really good filtering types, but I'd like to save them for progressive enhancement. Certainly food for thought. Cost could a taxonomy type, Free or Paid. This goes for the others as well, apart from rating which I'd like to have a star level system (which in Jekyll language might be just 1 to 5).

**Comment by CorstianBoerman** [comment]
I agree with @nouveller as I think a filtering system is really valuable! The point is that most of the time I'm searching for very specific types of projects. This makes me think of a project I did a few years ago. Here users could find their 'ultimate' car by going through a flow of questions and selecting what stuff they wanted. (eg, hatchback, diesel, max price of 20000, etc). If such thing is available on the website it'd be awesome. 

For example; find me a blog system which is actively beging developed, has theming and extension functionality, is written in some strange language I know and is free and open source.

In other words, I think (just my opinion) it's important to make sure all systems are described by all their properties which makes filtering like this possible.

**Comment by daviddarnes** [comment]
@CorstianBoerman wow that would be awesome! It'd be great to be able to filter multiple taxonomies at once to get just the right system. However due to the powers of Jekyll, plus the fact there isn't anything to filter, it be wasted development time at the moment. Not saying it isn't a great feature, I just don't want to spend time building this feature that won't yet be used to it's full potential.

For the time being using simple categories and custom taxonomy lists will be the most we'll need for now.

Note: This should totally be a feature request for when the site starts bulking out with content!

**Comment by daviddarnes** [comment]
### Updated sitemap
- **Front Page** *bread.li* - Some kind of featured simple header with the latest CMSs added in a list
- **CMSs** *bread.li/cmss/* - Basically an everything page, can click through types and things, multiple different taxonomy values
- **Submit** *bread.li/submit/* - Rules on submitting and what we need, then link through to GitHub contributing and Pull Requests
- **About** *bread.li/about/* - Explanation of the site and its purpose
- **Blog** *bread.li/blog/* - All those videos I've done, plus written content and updates
- **Jobs** *bread.li/jobs/* - Possibly a new collection where people can submit jobs as Pull Requests? Interesting idea.
